Role based authentication system #106

distalx opened the pull request:
#106 
**admin**
add-remove moderator. (role)

**moderator**
add moderator, volunteer (role)
block-unblock (user, volunteer)
edit-remove hangout. (any) (soft delete)
get notification of other users activity. (report, edit, delete, role update)
view app statistics. (online, active , blocked, new signups)

**volunteer**
edit hangout. (any)

**user**
create-edit-remove, join, hangout
create-edit-remove, upvote learning
**report** hangout (any)


The hierarchy flow  **admin** > **moderator** > **volunteer** > **user**.

lpatmo commented:
Amazing work, @distalx!

I made a commit to correct some minor wording. 

I QAed a couple of bugs:
- the profile page, FAQ page, About page, and hangout ID page aren't rendering the user profile image on the nav in the top right. I wonder if this is related to a missing subscription in routes.js. 
- The "notifications" page is blank, but the settings navbar says there's "1" notification:
![screen shot 2016-04-22 at 2 03 32 am](https://cloud.githubusercontent.com/assets/4512699/14736949/8d3ebc4a-082e-11e6-9c7c-dc340e7bc1bf.jpg)

I'll leave this PR up for a little while longer so other people can have a chance to code review. :)

`git checkout feature/auth-system-#106`

sergeant-q commented:
How do I test the various roles? Do I need to update my user profile to add a role to it? if so, what is the input/format for it? **admin** or??

distalx commented:
@lpatmo 
I have fixed the `rendering of the user profile image`.
I guess that notification you are getting is from previous built, I have changed the structure of notification collection that's why it's not displaying on the notification page. I'm sure that `meteor reset` will resolve that.

And I also bumped the meteor version , 1.3 takes much less time for build.
